near-sdk/src/store/lookup_set/mod.rs

```diff
@@ -105,7 +211,18 @@ where
         T: Borrow<Q>,
         Q: BorshSerialize + ToOwned<Owned = T> + Ord,
     {
-        self.map.remove(value).is_some()
+        let entry = self.get_mut_inner(&value);
+        match entry {
+            EntryState::Inserted => {
+                let _ = std::mem::replace(entry, EntryState::Absent);
+                true
+            }
+            EntryState::Present => {
+                let _ = std::mem::replace(entry, EntryState::Deleted);
```

Comment:
again, don't need `mem::replace` when you don't need the existing value. Can just assign the value